README.md

```diff
@@ -1,123 +1,343 @@
+
+
+
+### S3 Scanner
+
+To use AWS S3 as a source of input files, use AWS credentials as instructed below.
+
+Export the AWS access key, secret key, and credentials provider to environment variables. 
 
 ```
-export GCS_REID_QUERY_BUCKET=<name>
-cat << EOF | gsutil cp - gs://${REID_QUERY_BUCKET}/reid_query.sql
-${QUERY}
-EOF
+export AWS_ACCESS_KEY="<access_key>"
+export AWS_SECRET_KEY="<secret_key>"
+export AWS_CRED="{\"@type\":\"AWSStaticCredentialsProvider\",\"awsAccessKeyId\":\"${AWS_ACCESS_KEY}\",\"awsSecretKey\":\"${AWS_SECRET_KEY}\"}"
 ```
-Run the pipeline by passing required parameters:
+
+Use Gradle to build and run the job to performs data loss prevention (DLP) on a CSV file stored in Amazon S3. The results will be written to BigQuery.
+
 ```
-gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs="--region=<region> --project=<project_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=5 --maxNumWorkers=10 --runner=DataflowRunner --tableRef=<project_id>:<dataset>.<table> --dataset=<dataset> --topic=projects/<project_id>/topics/<name> --autoscalingAlgorithm=THROUGHPUT_BASED --workerMachineType=n1-highmem-4 --deidentifyTemplateName=projects/<project_id>/deidentifyTemplates/<name> --DLPMethod=REID --keyRange=1024 --queryPath=gs://<gcs_reid_query_bucket>/reid_query.sql"
+gradle spotlessApply
+
+gradle build
 
+// inspect is default as DLP Method; For deid: --DLPMethod=DEID
+gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs="--region=<region> --project=<project_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=1 --maxNumWorkers=2 --runner=DataflowRunner --filePattern=s3://<bucket>>/file.csv --dataset=<name>  --inspectTemplateName=<inspect_template> --deidentifyTemplateName=<deid_tmplate> --awsRegion=<aws_region> --awsCredentialsProvider=$AWS_CRED"
 ```
+
+#### Parameters:
+
+* --awsRegion: The region where the AWS resources reside.
+
+* --awsCredentialsProvider: The AWS credentials provider.
+
+
+## How to adapt this pipeline for your use cases
+
+1. [Create your own inspection templates & run inspection on sample data](https://cloud.google.com/dlp/docs/creating-templates-inspect)
+2. [Create de-identification templates and run de-identification on sample data](https://cloud.google.com/dlp/docs/creating-templates-deid)
+3. In case of cross-regional data source or sink, dataflow service account should be granted necessary permissions.
+
+
+## Troubleshooting
+
+Following are the known issues with Cloud DLP, along with ways you can avoid or recover from them.
+
+1. Duplicate Rows: When writing data to a BigQuery table, Cloud DLP might write duplicate rows.
+
+    > The project uses Streaming Inserts API of BigQuery which by default enables best-effort deduplication mechanism but it should not be relied upon as a mechanism to guarantee the absence of duplicates in your data.
```

Comment:
Let's make this a separate sentence and change it to active voice.

However, you should not rely on this mechanism to guarantee the absence of duplicates in your data.